.github/workflows/test-changed-firestore.yml

```diff
@@ -28,27 +34,115 @@ jobs:
           cp config/ci.config.json config/project.json
           yarn
       - name: build
-        run: yarn build:changed firestore
-      - name: Run tests if firestore or its dependencies has changed
-        run: yarn test:changed firestore
+        id: build
+        run: yarn build:changed firestore | egrep "Skipping all"
+        # Continue when "Skipping all" is not found
+        continue-on-error: true
+      - name: set output
+        # This means "Skipping all" was not found
+        if: steps.build.outcome != 'success'
+        id: set-output
+        run: echo "CHANGED=true" >> "$GITHUB_OUTPUT";
+      - name: Archive build
+        if: ${{ !cancelled() && steps.build.outcome != 'success' }}
+        run: |
+          tar -cf build.tar --exclude="\.git" .
+          gzip build.tar
+      - name: Upload build archive
+        if: ${{ !cancelled() && steps.build.outcome != 'success' }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: build.tar.gz
+          path: build.tar.gz
+          retention-days: ${{ env.artifactRetentionDays }}
+
+  compat-test-chrome:
+    name: Test Firestore Compatible
+
+    runs-on: ubuntu-latest
+
+    needs: build
+    if: ${{ needs.build.outputs.changed == 'true'}}
+    steps:
+      - name: Set up Node (14)
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14.x
+      - name: install Chrome stable
+        run: |
+          sudo apt-get update
+          sudo apt-get install google-chrome-stable
+      - name: Download build archive
+        uses: actions/download-artifact@v3
+        with:
+          name: build.tar.gz
+      - name: Unzip build artifact
+        run: tar xf build.tar.gz
+      - name: Bump Node memory limit
+        run: echo "NODE_OPTIONS=--max_old_space_size=4096" >> $GITHUB_ENV
+      - name: Test setup and yarn install
+        run: |
+          cp config/ci.config.json config/project.json
+      - name: Run compat tests
+        run: cd packages/firestore-compat && yarn run test:ci
+
+  test-chrome:
+    name: Test Firestore
+    strategy:
+      matrix:
+        test-name: ["test:browser", "test:travis", "test:lite:browser", "test:browser:prod:nameddb", "test:lite:browser:nameddb"]
```

Comment:
Do these tests work with `build:changed`? That is, can `build:changed` produce a subset of sources but these tests expect a full set of sources to be built?

Reply:
Yes, it appears so given that all tests pass with a firestore change.